lib/runner.js

```diff
@@ -47,36 +49,54 @@ Runner.prototype.addAfterHook = function (title, cb) {
 	this.tests.after.push(new Test(title, cb));
 };
 
-Runner.prototype.concurrent = function (tests) {
-	var self = this;
+Runner.prototype.addBeforeEachHook = function (title, cb) {
+	this.tests.beforeEach.push({
```

Comment:
I am not doing `new Test(title, cb)` here, because we need multiple instances of that test for `*each` hooks. So instead I am storing title and function and creating `Test` instances later.